packages/core/js-sdk/src/store/index.ts

```diff
@@ -5,13 +5,29 @@ import {
   SelectParams,
 } from "@medusajs/types"
 import { Client } from "../client"
-import { ClientHeaders } from "../types"
+import { ClientHeaders, Config } from "../types"
 
 export class Store {
   private client: Client
+  private config: Config
 
-  constructor(client: Client) {
+  constructor(client: Client, config: Config) {
     this.client = client
+    this.config = config
+  }
+
+  private withDefaultStoreHeaders(clientHeaders?: ClientHeaders) {
```

Comment:
The `client` has a `globalHeaders` field, so we can just set the `x-publishable-api-key` header there instead.

Basically, in the the `class Medusa` constructor just do
```
if (config.publishableKey) {
  config.globalHeaders['x-...'] = config.publishableKey
}
```

Reply:
won't that add the publishable key to all endpoints instead of just the store?

Reply:
It would, but it doesn't do any harm if you do, and you would rarely (if ever) call all of them from the same SDK instance.

But if you prefer this approach, I am happy to keep it as-is, it's just a bit more boilerplate, that's all

Reply:
Without a global header, external developers would have to pass this header themselves if they extend the SDK with more store endpoints right?

Reply:
yes, thats true. Reverted this!

Reply:
This would maybe be an argument for configuring this as a global header, as @sradevski suggested. Wdyt?

Reply:
I checked it, we already do add to the global headers. This change wasn't necessary

Reply:
![Uploading Screenshot 2024-09-11 at 11.01.08.png…]()

Reply:
![Screenshot 2024-09-11 at 11 01 08](https://github.com/user-attachments/assets/bff38621-1b87-4731-91a8-fec04610a609)

Reply:
Got it, nice!